cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java

```diff
@@ -246,6 +258,15 @@ public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
         } catch (Exception ex) {
             LOGGER.error("Error updating Stage Manager with status: {} reason: {}", status, ex);
         }
+
+        // We don't necessarily need to revive on every status update, but we might, and it's
+        // safe to do so.
+        reviveOffers();
```

Comment:
Also, I'm thinking out loud if sending out reviveOffers on each status update can be very chatty ? 

For ex, when launching a Cassandra node, you'll get an average of 3-5 TASK_RUNNING updates before it joins the ring. Each TASK_RUNNING update comes with `mode` information as part of data field. 

Doing back-of-envelope calculations suggests that the numbers aren't very bad for a single cluster with, for ex: 100 nodes, which comes to ~400 messages, which translates to 400 revive calls to Master. 

But, my worry is that it might quickly add up as folks run other services.